include/zephyr/drivers/sensor.h

```diff
@@ -106,6 +106,16 @@ enum sensor_channel {
 	/** Altitude, in meters */
 	SENSOR_CHAN_ALTITUDE,
 
+	/**
+	 * Particle count array
+	 * value1[(0.3-0.5µm)],
+	 * value2[(0.5-1.0µm)],
+	 * value3[(1.0-2.5µm)],
+	 * value4[(2.5-5.0µm)],
+	 * value5[(5.0-7.5µm)],
+	 * value6[(7.5-10.0µm)]
+	 */
+	SENSOR_CHAN_PARTICLE_COUNT,
```

Comment:
I've found sensors like the `DFRobot SEN0460` that has the same 6 values available (https://wiki.dfrobot.com/Gravity_PM2.5_Air_Quality_Sensor_SKU_SEN0460). But have also found sensor like the `SPS30` that just provides 5 count values (https://sensirion.com/media/documents/8600FF88/616542B5/Sensirion_PM_Sensors_Datasheet_SPS30.pdf).

Reply:
shouldn't this change be in a separate commit?

Reply:
At a minimum this should be a different commit, yes. 

But I am -1 on adding this new "array" channel as both your linked datasheets actually show that different sensors will likely use different "classes" of particle counts in this array i.e PM 0.5/1.0/2.5/4.0/10.0 for SPS30, and PM 0.3/0.5/1.0/2.5/5.0/10.0, none of which correspond to the ranges proposed here. HM3300/3600 sensors have yet another range (https://files.seeedstudio.com/wiki/Grove-Laser_PM2.5_Sensor-HM3301/res/HM-3300%263600_V2.1.pdf) and so on.

What would probably work though is adding `SENSOR_CHAN_PC_{X_Y}` channels.

Reply:
@kartben 
Thanks a lot for taking over and reviewing.

On first step I spent an own commit for this change, like you've requested.

And you are right, it's difficult, to generalize the particle count values for several sensors.

> What would probably work though is adding SENSOR_CHAN_PC_{X_Y} channels.

Could you give an example, how this would look like for my sngcja5 sensor with the PM 0.5/1.0/2.5/5.0/7.5/10.0 ? Haven't seen it in that way here before?

Reply:
> > What would probably work though is adding SENSOR_CHAN_PC_{X_Y} channels.
> 
> Could you give an example, how this would look like for my sngcja5 sensor with the PM 0.5/1.0/2.5/5.0/7.5/10.0 ? Haven't seen it in that way here before?

I mean just like you currently have `SENSOR_CHAN_PM_1_0`, `SENSOR_CHAN_PM_2_5`, etc. there could also be new channels for `SENSOR_CHAN_PC_0_3`, `SENSOR_CHAN_PC_0_5`, etc.?
Note that something like #63830 would probably make things easier here, but in the meantime I think adding channels for the common particle count sizes would work, but I will let @MaureenHelm and other core contributors to the sensor API confirm.

Reply:
Please go with this route today

Reply:
I'm not really seeing the need for this to be a part of sensor.h, why can't this highly specific 6 value channel be added in sngcja5.h? I would also like to add that with the new async API you're able to have an index along with a channel which would service this model much better. SENSOR_CHAN_PARTICLE_COUNT would be used along with an index to provide any number of dimensions to the measurement. We would still need a way to describe what each index means.